frontend/src/Context/index.tsx

```diff
@@ -0,0 +1,51 @@
+import { createContext, useReducer, Dispatch, ReactNode } from "react";
+
+interface QuickstartState {
+  linkSuccess: boolean;
+  isItemAccess: boolean;
+  linkToken: string | null;
+  accessToken: string | null;
+  itemId: string | null;
+  isError: boolean;
+}
+
+const initialState: QuickstartState = {
+  linkSuccess: false,
+  isItemAccess: false,
+  linkToken: "", //don't set to null or error message will show up briefly when site loads
+  accessToken: null,
+  itemId: null,
+  isError: false,
+};
+
+type QuickstartAction = {
+  type: "SET_STATE";
+  state: Partial<QuickstartState>;
+};
+
+interface QuickstartContext extends QuickstartState {
+  dispatch: Dispatch<QuickstartAction>;
+}
+
+const Context = createContext<QuickstartContext>(
+  initialState as QuickstartContext
+);
+
+const { Provider } = Context;
+export const QuickstartProvider: React.FC<{ children: ReactNode }> = (
+  props
+) => {
+  const reducer = (
+    state: QuickstartState,
+    action: QuickstartAction
+  ): QuickstartState => {
+    if (action.type === "SET_STATE") {
+      return { ...state, ...action.state };
+    }
+    return { ...state };
```

Comment:
usually a reducer is implemented as a switch statement.

```suggestion
  ): QuickstartState => {
switch (action.type) {
  case "SET_STATE":
    return { ...state, ...action.state };
    break;
  default:
    return { ...state };
}
```

Reply:
done